### PR TITLE
At/#1520 form transforming field rounding error

### PR DESCRIPTION
**Pull request by AlexeyTuboltsev**
make general data convertion from/to formik state on form load/ onSave.

### CODE REVIEW

File: app/components/eto/public-view/Cover.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ interface IProps {
   companyName: TTranslatedString;
   companyOneliner: TTranslatedString;
   companyLogo: IResponsiveImage;
-  tags: TTranslatedString[];
+  tags: TTranslatedString[] | string[] | undefined;
```
Comment (trival) [inline app/components/eto/public-view/Cover.tsx]: isn't TTranslatedString[] and string[] the same??

File: app/components/eto/registration/pages/InvestmentTerms.tsx [hunk 2]
```diff
@@ -41,7 +42,8 @@ interface IDispatchProps {
 
 type IProps = IExternalProps & IStateProps & IDispatchProps & FormikProps<TPartialEtoSpecData>;
 
-const EtoInvestmentTermsComponent: React.SFC<IProps> = ({ stateValues, savingData, readonly }) => {
+const EtoInvestmentTermsComponent: React.SFC<IProps> = p => {
```
Comment (trival) [inline app/components/eto/registration/pages/InvestmentTerms.tsx]: I like the old params destructuring better...

File: app/di/setupBindings.ts [hunk 3]
```diff
@@ -64,7 +64,7 @@ export function setupBindings(config: IConfig): Container {
   if (process.env.NODE_ENV === "production") {
     container.bind(symbols.logger).toConstantValue(noopLogger);
   } else {
-    container.bind<ILogger>(symbols.logger).toConstantValue(new DevConsoleLogger());
+    container.bind<ILogger>(symbols.logger).toConstantValue(noopLogger); //new DevConsoleLogger()); FIXME!!!!!
```
Comment (trival) [inline app/di/setupBindings.ts]: FIXME!!!!

File: app/components/eto/utils.ts [hunk 4]
```diff
@@ -25,3 +30,50 @@ export const applyDefaults = (data: any, defaults: any) => {
     return acc;
   }, dataCopy);
 };
+
+export const convert = (data: TPartialEtoSpecData, convertion: any) => {
```
Comment (trival) [inline app/components/eto/utils.ts]: I really like this approach! can we please add some unit tests for it?

Comment (trival) [reply app/components/eto/utils.ts]: also, could we name convertion into conversionSpec? Fixes typo, and is a little bit more clear...

File: app/components/shared/forms/form-field/FormTransformingField.tsx [hunk 5]
```diff
@@ -5,10 +5,10 @@ import * as React from "react";
 import { FormGroup, Input, InputGroup, InputGroupAddon } from "reactstrap";
 
 import { CommonHtmlProps, InputType } from "../../../../types";
+import { convertToPrecision } from "../../../eto/utils";
 import { FormLabel } from "./FormLabel";
-import { isNonValid, isValid } from "./utils";
-
 import * as styles from "./FormStyles.module.scss";
+import { isNonValid, isValid } from "./utils";
```
Comment (trival) [inline app/components/shared/forms/form-field/FormTransformingField.tsx]: could we fix imports?

File: app/components/shared/forms/form-field/FormTransformingField.tsx [hunk 6]
```diff
@@ -82,12 +65,11 @@ export class FormTransformingField extends React.Component<FieldGroupProps> {
                     <Input
                       className={cn(className, styles.inputField)}
                       {...field}
-                      value={transform(field.value, ratio) || ""}
-                      onChange={e => {
-                        setFieldTouched(name);
-                        setFieldValue(name, transformBack(e.target.valueAsNumber, ratio));
+                      onBlur={e => {
+                        setFieldValue(name, convertToPrecision(e.target.valueAsNumber, 2));
```
Comment (trival) [inline app/components/shared/forms/form-field/FormTransformingField.tsx]: If this component always does number precision transformation, can we rename it to FixedPrecisionFloatField or something, that explains what it does, so that it is clear when reading the code?

File: app/test-e2e/eto-registration/fixtures.ts [hunk 7]
```diff
@@ -67,9 +63,9 @@ export const legalInfoForm: TFormFixture = {
 
 // investment terms
 export const investmentTermsRequiredForm: TFormFixture = {
-  shareNominalValueEur: "1",
+  shareNominalValueEur: "3",
   preMoneyValuationEur: "1000000",
-  existingCompanyShares: "1000",
+  existingCompanyShares: "100",
```
Comment (Moejoe90) [inline app/test-e2e/eto-registration/fixtures.ts]: why is this changed to 100?
